apps/jobs/views.py

```diff
@@ -355,20 +365,20 @@ def leaderboard(request, challenge_phase_split_id):
 
     leaderboard_data = leaderboard_data.annotate(
         filtering_score=RawSQL('result->>%s', (default_order_by, ), output_field=FloatField())).values(
-            'id', 'submission__participant_team__team_name',
+            'id', 'submission__participant_team__team_name', 'submission__baseline_submission',
             'challenge_phase_split', 'result', 'filtering_score', 'leaderboard__schema', 'submission__submitted_at')
 
     sorted_leaderboard_data = sorted(leaderboard_data, key=lambda k: float(k['filtering_score']), reverse=True)
-
+    # challenge_host = challenge_phase_split__
     distinct_sorted_leaderboard_data = []
-    team_list = []
-
+    team_list = set()
```

Comment:
I don't think we need this? If yes, can you please explain why?

Reply:
`Set` is more efficient than using List for `x in s`.
List is `O(n)` whereas Set is `O(1)`

Reply:
The objects in set aren't ordered. whereas in list they are.

Reply:
But we need `team_list` only to check if data from a `team` is added to leaderboard or not.